src/GNDStk/Tree/test/read.test.cpp

```diff
@@ -3,49 +3,33 @@
 #include "GNDStk.hpp"
 
 SCENARIO("Testing GNDStk tree read() and operator>>") {
-   gnds::Tree<> t1, t2, t3, t4, t5;
+   GNDStk::Tree<> t1, t2, t3, t4;
 
    GIVEN("Some trees") {
-      // Note: there's no super compelling reason that our library actually
-      // has a direct (char *) version, as opposed to having only the string
-      // version and then relying on implicit conversion from char * to string.
-      // In fact, our library's (string) version just uses .c_str() and forwards
-      // to the (char *) version! In principle, we could make our (char *)
-      // version very slightly more efficient, by having it avoid conversions
-      // to string entirely. It doesn't do that right now, however, and, really,
-      // this is hardly an area in which we'd be concerned about performance.
-      // If our directly writing (char *) and (string) versions bothers anyone,
-      // we can easily chuck the (char *) variant.
-
-      WHEN("We read one from a char * filename") {
-         t1.read("n-008_O_016.xml");
-         REQUIRE(!t1.empty()); // we get something
-      }
-
       WHEN("We read one from a string filename") {
          const std::string filename = "n-008_O_016.xml";
-         t2.read(filename);
-         REQUIRE(!t2.empty()); // we get something
+         t1.read(filename);
+         REQUIRE(!t1.empty()); // we get something
```

Comment:
Is there something else we can test rather than just making we got "something"? Checking that we got "something" is a necessary, but not sufficient condition.

Reply:
Invariably, I think, the different portions of the Tree-test suite ended up having some overlap. For example, the *write* test calls read() too, because, well, we need to first create something (which is most easily done with read()) in order to write it. So, then, it turns out that there are other tests that end up effectively testing the read(), in the sense that they do their own things that ultimately depend on the read(). The result here was that this read() test ended up just being something that exercises the various read() functions. Basically, making sure they're still there. (So that if someone inadvertently removed one of the read() functions, the test wouldn't compile). I could certainly outfit this file with some similar material to what's in the write() function. Then it'll need to call write() in order to write the Tree contents and check against a gold standard. So, a proper write() test needs a read(), and a proper read() test needs a write()! And the distinction between the two blurs. :-)